src/utils/salesEvidenceFilter.ts

```diff
@@ -0,0 +1,59 @@
+import { SalesEvidenceFilterParams, SaleTypeEnum, UnprocessedResultsFromCRM, MinMaxNumberType, MinMaxDateType } from '../types'
+
+function numberFilter (property: UnprocessedResultsFromCRM, filterType: string, filterValues: MinMaxNumberType) {
+    return typeof property[filterType] === 'number' && property[filterType] >= filterValues.min && property[filterType] <= filterValues.max
+}
+
+function formatDateToString (date: Date): string {
+    const dateYear = date.getFullYear()
+    const monthValuesStartAtOne = 1
+    let dateMonth: string | number = date.getMonth() + monthValuesStartAtOne
+    let dateDate: string | number = date.getDate()
+    const numberToLeftPadZero = 9
+    if (dateMonth <= numberToLeftPadZero) dateMonth = `0${dateMonth}`
+    if (dateDate <= numberToLeftPadZero) dateDate = `0${dateDate}`
+    return `${dateYear}-${dateMonth}-${dateDate}`
+}
+
+function dateFilter (property: UnprocessedResultsFromCRM, dateSold: MinMaxDateType): boolean {
+    if (typeof dateSold.min !== 'undefined' && typeof dateSold.max !== 'undefined') {
+        const minDate = formatDateToString(dateSold.min)
+        const maxDate = formatDateToString(dateSold.max)
+        return typeof property.Sale_Date === 'string' && property.Sale_Date >= minDate && property.Sale_Date <= maxDate
+    }
+    return false
+}
+
+function saleTypeFilter (property: UnprocessedResultsFromCRM, saleTypes: SaleTypeEnum[]): boolean {
+    return saleTypes.some((saleType: SaleTypeEnum) => {
+        return property.Sale_Type.includes(saleType)
+    })
+}
+
+export default function salesEvidenceFilter (property: UnprocessedResultsFromCRM, filterParameters: SalesEvidenceFilterParams[]): boolean {
+    const {
+        landArea,
+        buildArea,
+        salePrice,
+        saleType,
+        dateSold
+    } = filterParameters[0]
```

Comment:
why item 0? Shouldn't this support adding multiple filters?

Reply:
Nope, as per Chris's specs. Check the attached video in the July 1st email.

Reply:
In that case I'd recommend not passing an array in. Code is confusing otherwise. If you don't want to do that, please add a comment.